documentation/flows/agent/dynamic_backhaul_switching.md

```diff
@@ -0,0 +1,113 @@
+
+---
+**NOTE**
+
+ The Wireless FSM state is out of scope for this design, since backhaul switching takes place in OPERATIONAL state alone - once the FSM is at the OPERATIONAL state, backhaul switching can be triggered in the backhaul manager on demand, and will result with FSM restart, which in turn will result with selecting a new backhaul.
+
+---
+
+## Requirements
+
+* Generic implementation in the backhaul manager which will provide a single entrypoint to perform backhaul switching from multiple clients
+* Supported clients - UCC, Controller, netlink events
+* Prevent loops - use only one backhaul at a time
+* Repeater static 
+
+## Assumptions
+
+* Wireless backhaul support for certification exists, including `dev_set_config,name,DUT,program,map,backhaul,RUID`
+* Repeater static configuration
+  * hostapds are configured with start_disabled=1 and static channel at boot
+    * the effect is that the AP will NOT hold a channel and will not put anything on-the-air
+  * Supplicants are configured for WPS_PBC - `wpa_cli wps_pbc multi_ap=1
+` in onboarding mode
+  * Supplicants are configured with credentials in operational mode
+
+# High Level Design
+
+Dynamic backhaul switching will be implemented as an API in the backhaul manager - `backhaul_manager_thread::switch_backhaul(const std::string iface_name)` which will move the backhaul manager FSM to the RESTART state, which also causes both radio agents restart.
+
+<a name="FSM_RESTART"></a>
+The backhaul manager FSM RESTART state will be updated to disable all backhaul interfaces by either removing them from the bridge, disabling them via netlink command or by adding an ebtable rule.
+
+Supported backhaul interfaces will be stored in UCI DB:
+
+* Wireless STA interfaces - `prplmesh.radio1.sta_iface='wlanX'`
+* Wired interface - `prplmesh.config.wired_backhaul='ethX'`
+
+---
+
+Currently, the wired interface is stored in config file - it is preferred to move it to UCI as a preparative commit to the dynamic backhaul switching feature.
+
+---
+
+In order to support automatic backhaul selection at init, another UCI configuration will be added to indicate the preferred backhaul type: `prplmesh.config.preferred_backhaul='wired/24g/5g/auto'`.
+
+A replication of the preferred backhaul type `m_preferred_backhaul` will be stored as a member variable in the backhaul manager thread which will be used for the dynamic backhaul switching, and can be overriden by clients requesting a backhaul switch.
+Note that in order to do a persistent switch, the client needs to also update the persistent UCI DB.
+
+---
+
+Note that `prplmesh.config.preferred_backhaul=auto` is out of scope for this design. It is expected that the client selecting the backhaul switch will set the specific backhaul band, which is the case with `dev_set_config` command and the controller Backhaul Steering Request.
```

Comment:
Ah, right, in that case, the preferred backhaul is needed of course... But I think specifying an interface is better.